### ISSUE TITLE
[Feature Request] Is it possible to have the logging evaluate "ONLY" if the logging level is active ?

### ISSUE THREAD

**Issue by Smurf-IV** [pocoproject/poco#4116]
According to the following code:
``` C++
	template <typename T, typename... Args>
	void fatal(const std::string& fmt, T arg1, Args&&... args)
	{
		log(Poco::format(fmt, arg1, std::forward<Args>(args)...), Message::PRIO_FATAL);
	}
```

The Format is always called for a logging line, and then the if condition to actually test if it should be "Logged" is done in the `log` function afterwards.

Would it not be more _performant_ to only call the logging formatting `iff` the level is "Useable" ?
i.e. move the checking before the formatting

**Comment by Smurf-IV** [comment]
Note: 
Yes I have just found out about the "convenience macros" Which are hidden in the code.
I think that they should be more prominent in the Documentation and in the test code examples please ?

**Comment by obiltschnig** [comment]
This has already been fixed in 1.11.7 and will be fixed in 1.12.5.
